File: Client/game-objects/wumpus.js
import MoveableObject from "./moveable-object.js";

export default class Wumpus extends MoveableObject {
	/**
	 * @param {number} x
	 * @param {number} y
	 */
	constructor(x, y) {
		super(x, y);

		this.#isAlive = true;
	}

	#isAlive = false;

	get isAlive() {
		return this.#isAlive;
	}

	die() {
		this.#isAlive = false;
	}
}

File: Server/HuntTheWumpus.API/Domain/GameObjects/Wumpus.cs
namespace HuntTheWumpus.API.Domain.GameObjects
{
    public class Wumpus : MoveableObject
    {
        public Wumpus(uint x, uint y) : base(x, y)
        {
            IsAlive = true;
        }

        public bool IsAlive { get; private set; }

        public void Die()
        {
            IsAlive = false;
        }
    }
}
